scripts/proto_splitter/proto2multi.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python3
+
+# Copyright 1996-2020 Cyberbotics Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author Simon Steinmann (https://github.com/Simon-Steinmann)
+#
+
+
+# Automatic PROTO to multi-file PROTO trimesh extractor
+#
+# INSTRUCTIONS:
+# Call the script with the --input=<path> argument
+#
+# 1. if <path> ends in <filename.proto>:
+#   - creates a new folder in the same directory named "multifile_<filename>"
+#   - creates a new <filename.proto> inside this folder with all trimeshes
+#     replaced by proto files, placed in a subfolder.
+#   - the mesh proto files in the subfolder have the same header as the
+#     original file, with the additional 'hidden' tag added.
+#
+# 2. If <path> does not end in ".proto" -> assumes directory
+#   - creates a backup folder next to our chosen <path>
+#   - searches for every .proto file in <path> recursively and:
+#       - ignores .proto file with either no header or a 'hidden' tag
+#       - replaces the file with a version, where meshes are extracted and put
+#         into a "<filename>_meshes" folder.
+#       - the mesh proto files in the subfolder have the same header as the
+#         original file, with the additional 'hidden' tag added.
+#
+#
+
+
+import os
+import optparse
+import shutil
+
+
+class proto2multi():
+    def __init__(self):
+        print('Proto 2 multi-file proto converter by Simon Steinmann')
+
+    def header(self, proto):
+        """Specify VRML file header."""
+        self.headerString += '# tags: hidden\n'
+        self.headerString += '# This is a proto file for Webots for the %s\n\n' % (self.robotName)
+        proto.write(self.headerString)
+
+    def createProto(self, string):
+        """turn mesh to proto file and stores it in the _meshes subfolder"""
+        name = self.robotName + 'Mesh' + str(self.shapeIndex)
+        filepath = '%s/%s.proto' % (self.meshFilesPath, name)
```

Comment:
This is a matter of a personal coding style, but I noticed that you combine `%` and `format()` to create strings.

https://docs.python.org/release/3.1.5/library/stdtypes.html#old-string-formatting-operations
`format()` is slightly more preferable in Python 2.6 and newer.